src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -182,25 +227,40 @@ public static RequestExecutor CreateForSingleNode(string url, string databaseNam
                 if (_disposed)
                     return false;
 
+                if (_isSingleNode) //precaution - if we are single node, no need to update topology
+                    return true;
+
                 using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
                 {
                     var command = new GetTopologyCommand();
 
                     await ExecuteAsync(node, context, command, shouldRetry: false).ConfigureAwait(false);
-
                     var serverHash = ServerHash.GetServerHash(node.Url, _databaseName);
 
                     TopologyLocalCache.TrySavingTopologyToLocalCache(serverHash, command.Result, context);
 
                     if (_nodeSelector == null)
-                        _nodeSelector = new NodeSelector(command.Result);
-
+                        _nodeSelector = GetNodeSelector(command.Result);
                     else if (_nodeSelector.OnUpdateTopology(command.Result))
                         DisposeAllFailedNodesTimers();
 
                     TopologyEtag = _nodeSelector.Topology.Etag;
                 }
             }
+            catch (Exception)
+            {
+                //since we failed to fetch topology, setup single-node temporary topology
+                //until we can retry fetching topology
+                if (_nodeSelector == null)
```

Comment:
Why? There is no way we can recover from this. Instead, we need to throw and we'll keep trying next time.